### PR TITLE
Added post on picker knob

### PR DESCRIPTION
**Pull request by anujmiddha**

### CODE REVIEW

File: _posts/2015-09-11-android-picker-knob.md [hunk 1]
```diff
@@ -0,0 +1,222 @@
+---
+layout: post
+title:  "Android Picker Knob"
+date:   2015-09-11 11:11:11
+author: anuj
+categories: Technical Android
+---
+
+In a game we are building, we needed a custom widget that looks like the volume control knob of old.
```
Comment (aashishdhawan) [inline _posts/2015-09-11-android-picker-knob.md]: what about `that looks like the classic (or old) volume control knob.`

File: _posts/2015-09-11-android-picker-knob.md [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+
+![Picker knob][screenshot]
+
+We went ahead and built a custom view for this. The view itself is available as a library [here][github-page]. Below is a brief description of the creation process.
```
Comment (aashishdhawan) [inline _posts/2015-09-11-android-picker-knob.md]: lets make `library` a link instead of `here`

File: _posts/2015-09-11-android-picker-knob.md [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+
+#### Custom or Compound view?
+Android provides a wide array of widgets that can be combined to create most UI elements. It is always a good idea to reuse existing widgets in a [Compound View][compound-view] if possible. For the picker knob, existing widgets were not ideal. Custom view it was then.
+
+#### Strategy
+
+__Mathematics__
+To get realistic physics and behavior, we visualized a strip of equidistant markers wound on a circular dial. Based on the current rotation, the view identifies the markers on the front half of the dial, projects them on a 1-d space (the lins AB in the diagram below) and draws them on the screen.
```
Comment (aashishdhawan) [inline _posts/2015-09-11-android-picker-knob.md]: `wound` ?

Comment (aashishdhawan) [inline _posts/2015-09-11-android-picker-knob.md]: `lins` ?

Comment (anujmiddha) [inline _posts/2015-09-11-android-picker-knob.md]: wound as in past tense of wind (to coil or twine about something)

lins is a type for line

File: _posts/2015-09-11-android-picker-knob.md (hunk 1 above)
Comment (anujmiddha) [inline _posts/2015-09-11-android-picker-knob.md]: I like it. Changed.

File: _posts/2015-09-11-android-picker-knob.md (hunk 2 above)
Comment (anujmiddha) [inline _posts/2015-09-11-android-picker-knob.md]: Done.

File: _posts/2015-09-11-android-picker-knob.md [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+            return true;
+
+        default:
+            endTouch(0);
+            return false;
+    }
+}
+```
+
+In startTouch, we will remove any previous running motions. In startScrollIfNeeded, we will check when the user has moved her finger by more than a threshold value, and then start processing the touch events. In processTouch, we will update the rotation as the user moves her finger. As soon as the user lifts the finger, we will calculate the velocity of the fling and give that velocity to our view. The complete code is available [here][picker-knob-java]
```
Comment (aashishdhawan) [inline _posts/2015-09-11-android-picker-knob.md]: we can use jekyll markdown to highlight keywords like `startScrollIfNeeded`

**Comment by aashishdhawan** [comment]
@anujmiddha done. Please merge when ready.

**Comment by aashishdhawan** [comment]
@anujmiddha Also please make sure you `squash` all these 4 commits in one clean commit `Added AndroidPickerKnob blog post (or something like that)` 
Keeps repo history clean. So every commit will be one blog added not some ugly looking commits.

**Pull request merged**